src/plugins/intel_cpu/src/nodes/kernels/stack_allocator.hpp

```diff
@@ -0,0 +1,478 @@
+#include <math.h>
+#include <dnnl_extension_utils.h>
+#include <cpu/x64/jit_generator.hpp>
+#include <stack>
+
+namespace ov {
+namespace intel_cpu {
+
+using namespace dnnl::impl;
+using namespace dnnl::impl::cpu;
+
+class StackAllocator final {
+public:
+    using Ptr = std::shared_ptr<StackAllocator>;
+
+    class Transaction;
+    class Address;
+    template<typename TReg>
+    class Reg;
+
+    StackAllocator(x64::jit_generator& code_gen)
+        : StackAllocator{code_gen, code_gen.rbp} {
+    }
+
+    StackAllocator(x64::jit_generator& code_gen, const Xbyak::Reg& bp)
+        : StackAllocator{code_gen, bp, 1} {
+    }
+
+    StackAllocator(x64::jit_generator& code_gen, const size_t alignment)
+        : StackAllocator{code_gen, code_gen.rbp, alignment} {
+    }
+
+    StackAllocator(x64::jit_generator& code_gen,
+                   const Xbyak::Reg& bp,
+                   const size_t alignment)
+        : code_generator{code_gen}
+        , base_pointer{Xbyak::Reg64{bp.getIdx()}}
+        , alignment{alignment} {
+        checkUnique(true);
+        alignStack(true);
+        code_generator.mov(base_pointer, code_generator.rsp);
+    }
+
+    ~StackAllocator() {
+        release();
+        alignStack(false);
+        checkUnique(false);
+    }
+
+    void release() {
+        current_offset = {};
+        commit();
+    }
+
+    friend void stack_mov(Address& addr, const Xbyak::Xmm& vmm);
+    friend void stack_mov(Address& addr, const Xbyak::Reg& reg);
+    friend void stack_mov(const Xbyak::Xmm& vmm, const Address& addr);
+    friend void stack_mov(const Xbyak::Reg& reg, const Address& addr);
+
+private:
+    struct Allocation {
+        using Ptr = std::shared_ptr<Allocation>;
+
+        Allocation(const Xbyak::Address& address,
+                   const size_t offset,
+                   const size_t size)
+           : address(address)
+           , offset(offset)
+           , size(size) {}
+
+        bool is_used = true;
+        bool is_transaction = false;
+        Xbyak::Address address;
+        size_t offset{};
+        size_t size;
+    };
+
+    void alignStack(bool isCtor) {
+        if (1 != alignment) {
+            constexpr size_t kReg64Size = 0x08;
+            if (isCtor) {
+                Xbyak::Label l_stack_aligned;
+                const Xbyak::Reg64 reg_base_stack_offset{base_pointer.getIdx()};
+                code_generator.xor_(reg_base_stack_offset, reg_base_stack_offset);
+                code_generator.mov(reg_base_stack_offset, static_cast<uint64_t>(kReg64Size));
+
+                const Xbyak::Reg64 reg_base_addr{Xbyak::Operand::RAX};
+                const Xbyak::Reg64 reg_reminder{Xbyak::Operand::RDX};
+                const Xbyak::Reg64 reg_alignment{Xbyak::Operand::RCX};
+
+                code_generator.push(reg_base_addr);
+                code_generator.push(reg_reminder);
+                code_generator.push(reg_alignment);
+
+                code_generator.xor_(reg_base_addr, reg_base_addr);
```

Comment:
The `xor_` here is possibly redundant also.

Reply:
Removed